src/Components/CreateImageWizard/formComponents/RepositoriesStatus.tsx

```diff
@@ -1,62 +1,139 @@
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  Popover,
+} from '@patternfly/react-core';
 import {
   CheckCircleIcon,
   ExclamationCircleIcon,
   ExclamationTriangleIcon,
+  ExternalLinkAltIcon,
   InProgressIcon,
 } from '@patternfly/react-icons';
 
 import { ApiRepositoryResponse } from '../../../store/contentSourcesApi';
+import {
+  convertStringToDate,
+  timestampToDisplayString,
+} from '../../../Utilities/time';
+import { useGetEnvironment } from '../../../Utilities/useGetEnvironment';
+
+const getLastIntrospection = (
+  repoIntrospections: RepositoryStatusProps['repoIntrospections']
+) => {
+  const currentDate = Date.now();
+  const lastIntrospectionDate = convertStringToDate(repoIntrospections);
+  const timeDeltaInSeconds = Math.floor(
+    (currentDate - lastIntrospectionDate) / 1000
+  );
+
+  if (timeDeltaInSeconds <= 60) {
+    return 'A few seconds ago';
+  } else if (timeDeltaInSeconds <= 60 * 60) {
+    return 'A few minutes ago';
+  } else if (timeDeltaInSeconds <= 60 * 60 * 24) {
+    return 'A few hours ago';
+  } else {
+    return timestampToDisplayString(repoIntrospections);
+  }
+};
 
 type RepositoryStatusProps = {
   repoStatus: ApiRepositoryResponse['status'];
   repoUrl: ApiRepositoryResponse['url'];
+  repoIntrospections: ApiRepositoryResponse['last_introspection_time'];
+  repoFailCount: ApiRepositoryResponse['failed_introspections_count'];
 };
 
-const RepositoriesStatus = ({ repoStatus, repoUrl }: RepositoryStatusProps) => {
+const RepositoriesStatus = ({
+  repoStatus,
+  repoUrl,
+  repoIntrospections,
+  repoFailCount,
+}: RepositoryStatusProps) => {
+  const { isBeta } = useGetEnvironment();
   if (repoStatus === 'Valid') {
     return (
       <>
         <CheckCircleIcon className="success" /> {repoStatus}
       </>
     );
-  } else if (repoStatus === 'Invalid') {
-    return (
-      <>
-        <Popover
-          position="bottom"
-          minWidth="30rem"
-          bodyContent={
-            <>
-              <Alert variant="danger" title="Invalid" isInline isPlain />
-              Cannot fetch {repoUrl}
-            </>
-          }
-        >
-          <Button variant="link" className="pf-u-p-0 pf-u-font-size-sm">
-            <ExclamationCircleIcon className="error" />{' '}
-            <span className="failure-button">{repoStatus}</span>
-          </Button>
-        </Popover>
-      </>
-    );
-  } else if (repoStatus === 'Unavailable') {
+  } else if (repoStatus === 'Invalid' || repoStatus === 'Unavailable') {
     return (
       <>
         <Popover
           position="bottom"
           minWidth="30rem"
           bodyContent={
             <>
-              <Alert variant="warning" title="Unavailable" isInline isPlain />
-              Cannot fetch {repoUrl}
+              <Alert
+                variant={repoStatus === 'Invalid' ? 'danger' : 'warning'}
+                title={repoStatus === 'Invalid' ? 'Invalid' : 'Unavailable'}
```

Comment:
Because we check up to that that `repoStatus` can be either `Invalid` or `Unavailable` then you can just let the title be `repoStatus`

Reply:
Yup :no_mouth:, thanks for pointing that out :sweat_smile: